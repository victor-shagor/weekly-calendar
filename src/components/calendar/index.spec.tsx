import { render, screen } from "@testing-library/react";
import "@testing-library/jest-dom/extend-expect";
import renderer from "react-test-renderer";
import Calendar from ".";

describe("days view", () => {
  it("should render Calendar view", () => {
    render(<Calendar />);
    const CalendarDiv = screen.getByTestId("calender");

    expect(CalendarDiv.querySelector(".btn")).toBeInTheDocument();
    expect(CalendarDiv.querySelectorAll(".btn").length).toEqual(2);
    expect(CalendarDiv.querySelector(".btn-date")).toBeInTheDocument();
  });

  it("renders correctly", () => {
    const tree = renderer.create(<Calendar />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
