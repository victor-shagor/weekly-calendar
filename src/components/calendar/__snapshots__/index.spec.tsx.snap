// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`days view renders correctly 1`] = `
<div
  className="wrapper"
  data-testid="calender"
>
  <div
    className="calendar-container"
  >
    <div
      className="calendar-header"
      data-testid="header"
    >
      <h2>
        October-2022
      </h2>
      <div>
        <button
          className="btn"
          onClick={[Function]}
          type="button"
        >
          <svg
            enableBackground="new 0 0 48 48"
            fill="currentColor"
            height={25}
            stroke="currentColor"
            strokeWidth="0"
            style={
              Object {
                "color": undefined,
              }
            }
            version="1"
            viewBox="0 0 48 48"
            width={25}
            xmlns="http://www.w3.org/2000/svg"
          >
            <polygon
              fill="#2196F3"
              points="30.9,43 34,39.9 18.1,24 34,8.1 30.9,5 12,24"
            />
          </svg>
        </button>
        <button
          className="btn"
          onClick={[Function]}
          type="button"
        >
          <svg
            enableBackground="new 0 0 48 48"
            fill="currentColor"
            height={25}
            stroke="currentColor"
            strokeWidth="0"
            style={
              Object {
                "color": undefined,
              }
            }
            version="1"
            viewBox="0 0 48 48"
            width={25}
            xmlns="http://www.w3.org/2000/svg"
          >
            <polygon
              fill="#2196F3"
              points="17.1,5 14,8.1 29.9,24 14,39.9 17.1,43 36,24"
            />
          </svg>
        </button>
      </div>
    </div>
    <div
      data-testid="daysview-container"
    >
      <div
        style={
          Object {
            "display": "flex",
            "justifyContent": "space-between",
            "margin": "1rem",
          }
        }
      >
        <div>
          S
        </div>
        <div>
          M
        </div>
        <div>
          T
        </div>
        <div>
          W
        </div>
        <div>
          T
        </div>
        <div>
          F
        </div>
        <div>
          S
        </div>
      </div>
      <div
        data-testid="day-grid"
        style={
          Object {
            "display": "flex",
            "justifyContent": "space-between",
            "margin": "0.3rem",
          }
        }
      >
        <div>
          <button
            className="btn-date"
            onClick={[Function]}
            type="button"
          >
            <time
              dateTime="2022-10-23"
            >
              23
            </time>
          </button>
        </div>
        <div>
          <button
            className="btn-date"
            onClick={[Function]}
            type="button"
          >
            <time
              dateTime="2022-10-24"
            >
              24
            </time>
          </button>
        </div>
        <div>
          <button
            className="btn-date"
            onClick={[Function]}
            type="button"
          >
            <time
              dateTime="2022-10-25"
            >
              25
            </time>
          </button>
        </div>
        <div>
          <button
            className="btn-date"
            onClick={[Function]}
            type="button"
          >
            <time
              dateTime="2022-10-26"
            >
              26
            </time>
          </button>
        </div>
        <div>
          <button
            className="blue btn-date"
            onClick={[Function]}
            type="button"
          >
            <time
              dateTime="2022-10-27"
            >
              27
            </time>
          </button>
        </div>
        <div>
          <button
            className="btn-date"
            onClick={[Function]}
            type="button"
          >
            <time
              dateTime="2022-10-28"
            >
              28
            </time>
          </button>
        </div>
        <div>
          <button
            className="btn-date"
            onClick={[Function]}
            type="button"
          >
            <time
              dateTime="2022-10-29"
            >
              29
            </time>
          </button>
        </div>
      </div>
    </div>
  </div>
</div>
`;
